src/v/cluster/types.h

```diff
@@ -4418,12 +4418,12 @@ struct topic_disabled_partitions_set
     bool is_disabled(model::partition_id id) const {
         return !partitions || partitions->contains(id);
     }
-    bool is_topic_disabled() const { return !partitions.has_value(); }
-    bool is_empty() const { return partitions && partitions->empty(); }
+    bool is_fully_disabled() const { return !partitions.has_value(); }
```

Comment:
a general question, for my understanding, is there a difference between  partitions=std::nullopt and partitions.empty() = true?  Curious if we really need an optional here.

Reply:
The difference is explained in the comment a few lines earlier - partitions=nullopt means *all* partitions are disabled. It looks a bit awkward but I thought that this is better than having a separate "all disabled" flag.